### PR TITLE
fix(crons): Validate check-in duration

### PR DESCRIPTION
**Pull request by rjo100**
Validates + logs invalid check-in durations

Closes SENTRY-114M
Closes SENTRY-114T
Closes SENTRY-114P
Closes SENTRY-114S
Closes SENTRY-11AV
Closes SENTRY-114W
Closes SENTRY-107X
Closes SENTRY-11A3
Closes SENTRY-11A0
Closes SENTRY-119R
Closes SENTRY-118X
Closes SENTRY-1193
Closes SENTRY-118E
Closes SENTRY-118D
Closes SENTRY-114Z
Closes SENTRY-114K
Closes SENTRY-10WM
Closes SENTRY-114J
Closes SENTRY-107G
Closes SENTRY-114D
Closes SENTRY-114C
Closes SENTRY-114A
Closes SENTRY-1144
Closes SENTRY-10J8
Closes SENTRY-112M
Closes SENTRY-112H
Closes SENTRY-111X
Closes SENTRY-111W
Closes SENTRY-111V

### CODE REVIEW

File: src/sentry/monitors/consumers/monitor_consumer.py [hunk 1]
```diff
@@ -165,6 +174,14 @@ def _process_message(wrapper: Dict) -> None:
                 if duration is None:
                     duration = int((start_time - check_in.date_added).total_seconds() * 1000)
 
+                if not valid_duration(duration):
+                    metrics.incr(
+                        "monitors.checkin.result",
+                        tags={**metric_kwargs, "status": "failed_duration_check"},
+                    )
+                    logger.debug("check-in duration is invalid: %s", project.organization_id)
```
Comment (rjo100) [inline src/sentry/monitors/consumers/monitor_consumer.py:182]: not sure if i wanted to move this logic to the helper method, but then i'd have the pass the params which I don't really want to do. long term we should have this logic run through the validator

File: src/sentry/monitors/validators.py [hunk 2]
```diff
@@ -223,7 +224,12 @@ class MonitorCheckInValidator(serializers.Serializer):
             ("in_progress", CheckInStatus.IN_PROGRESS),
         )
     )
-    duration = EmptyIntegerField(required=False, allow_null=True)
+    duration = EmptyIntegerField(
+        required=False,
+        allow_null=True,
+        max_value=BoundedPositiveIntegerField.MAX_VALUE,
```
Comment (evanpurkhiser) [inline src/sentry/monitors/validators.py:230]: Do we do this type of thing anywhere else?

Comment (rjo100) [reply src/sentry/monitors/validators.py:230]: A couple places but we usually just type out `2147483647` and stick it in the file

Comment (rjo100) [reply src/sentry/monitors/validators.py:230]: I guess since this is a static number probably OK to copy instead of reference

Comment (evanpurkhiser) [reply src/sentry/monitors/validators.py:230]: Nah I'm fine with the constant, was just curious if this was our usual pattern

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**